### PR TITLE
Catch SIGINT (Ctrl-c) and exit cleanly

### PR DESCRIPTION
**Pull request by loverend**
I am adding signal to catch `SIGINT` in `cli.py` and not send it to the
child processes. When a `SIGINT` is caught, `cli` then calls the
`stop_engine` funcion in `napalm_log`, which sends a SIGTERM to each
child process. The child process will catch this `SIGTERM` and run it's
own `stop` function, while will close all pipes / sockets, and terminate
the loop.

### CODE REVIEW

File: napalm_logs/auth.py [hunk 1]
```diff
@@ -124,11 +129,18 @@ def start(self):
             except ssl.SSLError:
                 log.exception('SSL error', exc_info=True)
                 continue
+            except socket.error:
+                if self.__up is False:
+                    break
+                else:
+                    raise
```
Comment (mirceaulinic) [inline napalm_logs/auth.py]: Please let's raise and catch later the `NapalmLogsExit` exception (https://github.com/napalm-automation/napalm-logs/blob/master/napalm_logs/exceptions.py#L47) -- unused yet, but defined exactly for this scope :)

File: napalm_logs/auth.py [hunk 2]
```diff
@@ -124,11 +129,18 @@ def start(self):
             except ssl.SSLError:
                 log.exception('SSL error', exc_info=True)
                 continue
+            except socket.error:
+                if self.__up is False:
+                    break
```
Comment (mirceaulinic) [inline napalm_logs/auth.py]: Please raise or return

File: napalm_logs/device.py [hunk 3]
```diff
@@ -285,11 +290,18 @@ def start(self):
...
         self.__up = True
         while self.__up:
             # bin_obj = self.sub.recv()
             # msg_dict, address = umsgpack.unpackb(bin_obj, use_list=False)
-            msg_dict, address = self.pipe.recv()
+            try:
+                msg_dict, address = self.pipe.recv()
+            except IOError:
+                if self.__up is False:
+                    break
```
Comment (mirceaulinic) [inline napalm_logs/device.py]: Same comments here: return and/or raise `NapalmLogsExit`

File: napalm_logs/server.py [hunk 4]
```diff
@@ -116,12 +121,19 @@ def start(self):
...
         while self.__up:
             # Take messages from the main queue
             # bin_obj = self.sub.recv()
             # msg, address = umsgpack.unpackb(bin_obj, use_list=False)
-            msg, address = self.pipe.recv()
+            try:
+                msg, address = self.pipe.recv()
+            except IOError:
+                if self.__up is False:
+                    break
```
Comment (mirceaulinic) [inline napalm_logs/server.py]: Also here please

**Comment by loverend** [comment]
Addressed all comments and updated.

**Pull request merged**